ReactAndroid/src/main/java/com/facebook/react/devsupport/DevSupportManagerImpl.java

```diff
@@ -522,6 +532,15 @@ private void resetCurrentContext(@Nullable ReactContext reactContext) {
       mDebugOverlayController = new DebugOverlayController(reactContext);
     }
 
+    if (mDevSettings.isHotModuleReplacementEnabled() && mCurrentContext != null) {
+      try {
+        URL sourceUrl = new URL(getSourceUrl());
+        mCurrentContext.getJSModule(HMRClient.class).enable("android", sourceUrl.getPath().substring(1));
+      } catch (Exception e) {
```

Comment:
Catch only the exceptions you want to catch (you can | the class names together in the catch statement if you want to handle them all in the same way)